frontend/service.go

```diff
@@ -1024,7 +1024,7 @@ func (s *DarksideStreamer) ClearIncomingTransactions(ctx context.Context, e *wal
 func (s *DarksideStreamer) AddAddressUtxo(ctx context.Context, arg *walletrpc.GetAddressUtxosReply) (*walletrpc.Empty, error) {
 	utxosReply := common.ZcashdRpcReplyGetaddressutxos{
 		Address:     arg.Address,
-		Txid:        hash32.Encode(hash32.Reverse(hash32.T(arg.Txid))),
+		Txid:        hash32.Encode(hash32.Reverse(hash32.FromSlice(arg.Txid))),
```

Comment:
I'm confused about `hash32.Encode` - what does this do that is different from `hex.EncodeToString` as it's used directly below? It might be a good idea to define a `TxId.Encode` method or something of the sort  (and the dual, the decoding method) so that this low-level transformation isn't done at the use site and is guaranteed to be done consistently everywhere, as the encoding has caused problems in the past.